lib/consolidate.js

```diff
@@ -893,12 +896,16 @@ exports.underscore = fromStringRenderer('underscore');
 exports.underscore.render = function(str, options, cb) {
   return promisify(cb, function(cb) {
     var engine = requires.underscore || (requires.underscore = require('underscore'));
+    var extend = (requires.extend || (requires.extend = require('util')._extend));
     try {
+      const partials = {};
       for (var partial in options.partials) {
-        options.partials[partial] = engine.template(options.partials[partial]);
+        partials[partial] = engine.template(options.partials[partial]);
       }
-      var tmpl = cache(options) || cache(options, engine.template(str, null, options));
-      cb(null, tmpl(options).replace(/\n$/, ''));
+      const optionsCopy = extend({}, options);
```

Comment:
I don't think making a copy of `options` is necessary here since the copy was already made in `fromStringRenderer` above and already has the new `partials` object.